lib/sequelize.js

```diff
@@ -577,11 +575,11 @@ class Sequelize {
               return this.connectionManager.releaseConnection(connection);
             }
           });
-      }), retryOptions)).finally(() => {
+      }).finally(() => {
```

Comment:
this block can be removed then, merge it with `finally` block just above it which is releasing connection. decrement then release connection